crates/forge/src/cheatcodes_hint_processor.rs

```diff
@@ -400,20 +409,42 @@ fn execute_syscall(
     system: &ResOperand,
     vm: &mut VirtualMachine,
     cheatnet_state: &mut CheatnetState,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &Box<dyn Any>,
+    constants: &HashMap<String, Felt252>,
+    original_cairo_hint_processor: &mut OriginalCairoHintProcessor,
 ) -> Result<(), HintError> {
     let (cell, offset) = extract_buffer(system);
-    let system_ptr = get_ptr(vm, cell, &offset)?;
-
-    let mut buffer = MemBuffer::new(vm, system_ptr);
-
-    let selector = buffer.next_felt252().unwrap().to_bytes_be();
-
-    if std::str::from_utf8(&selector).unwrap() != "CallContract" {
-        return Err(HintError::CustomHint(Box::from(
-            "starknet syscalls cannot be used in tests".to_string(),
-        )));
-    }
+    let mut system_ptr = get_ptr(vm, cell, &offset)?;
+
+    // We peek into memory to check the selector
+    let selector = DeprecatedSyscallSelector::try_from(felt_to_stark_felt(
+        &vm.get_integer(system_ptr).unwrap(),
+    ))?;
+    system_ptr.offset += 1;
```

Comment:
Why do we increment it here 🤔  I think it will break if you call another syscall after keccak syscall

Reply:
Why do you think so? In the last test, I call the Keccack syscall and then the CallContract syscall. Everything works fine

Reply:
I think it's fine since `get_integer` probably does not change the `offset`, but couldn't we just use `next_felt252` which (I assume) increments `offset` under the hood?

Reply:
We can't do it, because we need to pass the VM to the `original_cairo_hint_processor.execute_hint` which would not be possible if we created the buffer before.

Reply:
Makes sense, we do not want to clone the vm